app/serializers/api/v3/unite_legale_serializer.rb

```diff
@@ -8,6 +8,6 @@ class API::V3::UniteLegaleSerializer < ApplicationSerializer
   attribute :etablissement_siege
 
   def etablissement_siege
-    object.etablissements.where(etablissement_siege: 'true').first
+    object.etablissements.where(etablissement_siege: ['true', 't']).order(nil).limit(1).first
```

Comment:
Je ne comprends pas le `order` ni le `limit` ici.

Reply:
Le `order(nil)` permet de supprimer le order sur les ids qu'ont a identifié comme étant un bottleneck de perf (en le retirant on passe de 10s à <1s de résolution sur certains sirets..)

Le `limit` force aussi la requête SQL, je suis pas sûr que ça soit vraiment nécessaire cela dit.

Il s'agit d'un "hotfix" pour éviter de bloquer sur certains sirets, il faudrait revoir le schema et les index dans un second temps (on a noté ça dans le basecamp interne btw).

C'est pas ouf, mais ça permet de résoudre tout de suite ce problème

Reply:
Je vais ajouter un commentaire

Reply:
J'avais pas vu passer le doc de l'étude sur le basecamp, je comprends mieux 👍🏼